### PR TITLE
Use `Did` for json output and other places

### PR DESCRIPTION
**Pull request by cloudhead**
Instead of showing just public keys we need to show the fully qualified spec for a NID which is prefixed by `did:key:` While we want to maintain `Author` and `ActorId` for internal use.

### CODE REVIEW

File: radicle-cli/examples/rad-id.md [hunk 1]
```diff
@@ -136,7 +136,7 @@ Our key was added to the `Rejected` set of `keys` and the `total`
 increased to `1`.
 
     Rejected
-    
+
     total: 1
     keys: [
       "z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi"
```
Comment (FintanH) [inline radicle-cli/examples/rad-id.md:142]: Maybe this should be `"delegates": [ "did:key:<...>", ...]`

File: radicle-cli/src/commands/patch/list.rs [hunk 2]
```diff
@@ -85,7 +85,7 @@ fn print(
 ) -> anyhow::Result<()> {
     let target_head = common::patch_merge_target_oid(patch.target(), storage)?;
 
-    let you = patch.author().id() == whoami;
+    let you = &**patch.author().id() == whoami;
```
Comment (FintanH) [inline radicle-cli/src/commands/patch/list.rs:88]: suggestion: have a method on `Did` for checking if a `PublicKey` is owned by that `Did`

File: radicle/src/cob/common.rs [hunk 3]
```diff
@@ -10,19 +10,25 @@ pub use radicle_crdt::clock::Physical as Timestamp;
...
+        Self { id: id.into() }
     }
 
-    pub fn id(&self) -> &NodeId {
+    pub fn id(&self) -> &Did {
         &self.id
     }
 }
 
+impl From<NodeId> for Author {
```
Comment (FintanH) [inline radicle/src/cob/common.rs:26]: Might make more sense to use `PublicKey` since `NodeId` is just an alias.

**Review by FintanH — APPROVED** [review]
Generally looks good and left a few suggestions :)

File: radicle-cli/examples/rad-id.md (hunk 1 above)
Comment (cloudhead) [reply radicle-cli/examples/rad-id.md:142]: Was thinking the same yes.

File: radicle-cli/src/commands/patch/list.rs (hunk 2 above)
Comment (cloudhead) [reply radicle-cli/src/commands/patch/list.rs:88]: Hmmmm, I think the "ownership" relation might be a bit weird and come out of nowhere. I'd rather have maybe a `as_key()` method.

Comment (FintanH) [reply radicle-cli/src/commands/patch/list.rs]: But that's essentially what DIDs boil down to in a lot of cases, key ownership (and then associating actions across uses of those keys). But `as_key` also works, just goes in the opposite direction :)

Comment (cloudhead) [reply radicle-cli/src/commands/patch/list.rs]: Yeah, it's just that right now, our `Did` type is basically `did:key`, so I find it a bit confusing to use the verb "owns" when they're kind of the same thing..

**Pull request merged**